internal/service/alarms/internal/alertmanager/alertmanager.go

```diff
@@ -1,29 +1,41 @@
 package alertmanager
 
 import (
+	"bytes"
 	"context"
 	_ "embed"
 	"fmt"
 	"log/slog"
+	template "text/template"
 
 	corev1 "k8s.io/api/core/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift-kni/oran-o2ims/internal/controllers/utils"
 )
 
 const (
-	namespace  = "open-cluster-management-observability"
 	secretName = "alertmanager-config"
 	secretKey  = "alertmanager.yaml"
 )
 
-//go:embed alertmanager.yaml
+const templateName = "alertmanager.yaml"
+
+//go:embed alertmanager.yaml.template
 var alertManagerConfig []byte
 
 // Setup updates the alertmanager config secret with the new configuration
 func Setup(ctx context.Context, cl client.Client) error {
+	// Get the ACM namespace
+	acmNamespace, err := utils.GetACMNamespace(ctx, cl)
+	if err != nil {
+		return fmt.Errorf("failed to get ACM namespace: %w", err)
+	}
+	namespace := fmt.Sprintf("%s-observability", acmNamespace)
```

Comment:
think its better to just hardcode this ACM namespace? might be worth decoupling from searchAPI i think

Reply:
The problem is that the AMC namespace isn't fixed.  In my lab it is `open-cluster-management` and in another lab it is `rhacm`.  I figured (maybe incorrectly) that the alertmanager would be in whatever namespace the search-api is in + the `-observability` suffix.

Reply:
> ACM namespace isn't fixed

tbh i think its pretty much guaranteed to be fixed if done through official channel. And for observability stack if `open-cluster-management-observability` doesn't exist the whole operator may actually fail

Interesting that you have a ns `rhacm`! Like if we dont have `open-cluster-management` and `open-cluster-management-observability` probably a symptom of bigger problem (acm and observability not installed correctly)!

Reply:
I agree, this is clearly stated in the documentation. Let's not try to support other options. Our system should expect everything to be in `open-cluster-management-observability`

Reply:
can you point me to the documentation?

Reply:
https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.11/html/observability/observing-environments-intro#configuring-alertmanager

Reply:
The documentation never says that the namespace MUST be a specific value.  It simply walks through an example of how to configure it and it happens to have chosen that as a name.    Having said that, if both of you feel that hardcoding it is an acceptable strategy then we can certainly do that.